### PR TITLE
Take 2 on auto figure uploads

### PR DESCRIPTION
**Pull request by Cadair**
This time from a branch on sunpy/sunpy so I can get the ssh key and test it.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -72,7 +76,6 @@ jobs:
         name: py36_test_oldestdeps
 
       - linux: py38-devdeps
-        name: py38_test_devdeps
```
Comment (nabobalis) [inline azure-pipelines.yml:75]: why no name?

File: .circleci/config.yml [hunk 2]
```diff
@@ -145,25 +89,76 @@ jobs:
...
+      - run: git clone git@github.com:sunpy/sunpy-figure-tests.git -b sunpy-test ~/sunpy-figure-tests/
+
+      - run: cp -r .tmp/figure_py36/figure_test_images/*.png ~/sunpy-figure-tests/figures/figure_py36/
+
+      - run: cp -r .tmp/figure_py37_astropydev/figure_test_images/*.png ~/sunpy-figure-tests/figures/figure_py37_astropydev/
+
+      - run: git config --global user.email "sunpy@circleci" && git config --global user.name "SunPy Circle CI"
+
+      - run: cd ~/sunpy-figure-tests/; git add . && git commit -m "Update reference figures from ${CIRCLE_BRANCH}" && git push
+
```
Comment (nabobalis) [inline .circleci/config.yml:139]: Why all these line breaks?

File: azure-pipelines.yml (hunk 1 above)
Comment (Cadair) [reply azure-pipelines.yml:75]: it doesn't need a name because we don't reference it in the upload task later.

Comment (nabobalis) [reply azure-pipelines.yml:75]: But it won't have a nice name?

Comment (Cadair) [reply azure-pipelines.yml:75]: the name is not used anywhere else iirc

File: .circleci/config.yml [hunk 3]
```diff
@@ -145,25 +89,83 @@ jobs:
...
+      - deploy-reference-images:
+          requires:
+            - "figure_py36"
+            - "figure_py37_astropydev"
+          filters:
+            branches:
+              only:
+                - master
+                # - 1.1
+                # - 1.0
```
Comment (nabobalis) [inline .circleci/config.yml:167]: Will this be enabled?

File: azure-pipelines.yml (hunk 1 above)
Comment (nabobalis) [reply azure-pipelines.yml:75]: Is it not in the GUI?

Comment (Cadair) [reply azure-pipelines.yml:75]: No, I don't think so the template generates that from the URL.and tox env iirc

File: .circleci/config.yml (hunk 3 above)
Comment (Cadair) [reply .circleci/config.yml:167]: Yeah, we need to create the branch on the repo first and then back port this.

**Comment by Cadair** [comment]
Need to fix the figure comparison page I think.

**Comment by dstansby** [comment]
I've unsubscribed from this pr, but please do ping me when it's ready for review

**Comment by Cadair** [comment]
@dstansby I think this is ready for review. I am not 100% sure it's working, but I can't really tell until it's on master now.

**Comment by nabobalis** [comment]
You want to remove that blank test file and the figure comparison can't get the old images (I assume its a non master branch issue?)

File: azure-pipelines.yml [hunk 4]
```diff
@@ -72,7 +79,6 @@ jobs:
...
```
Comment (dstansby) [inline azure-pipelines.yml:75]: Why has this gone?

Comment (Cadair) [reply azure-pipelines.yml:75]: See https://github.com/sunpy/sunpy/pull/4004#discussion_r405604079 TL;DR it isn't used for anything.

**Comment by Cadair** [comment]
The wheels/sdists are broken because of the PIL issue, and we don't have the same pinning things. They are only showing up here because this PR is a branch on sunpy/sunpy

**Pull request merged**